MobileWallet/Backup/ICloudBackup.swift

```diff
@@ -231,81 +450,45 @@ extension ICloudBackup {
-            self?.notifyObservers(percent: 0, completed: false, error: nil)
-            self?.inProgress = true
-            self?.progressValue = 0.0
-            self?.processCloudFiles()
-        }
-
-        NotificationCenter.default.addObserver(forName: NSNotification.Name.NSMetadataQueryDidUpdate, object: query, queue: query.operationQueue) { [weak self] (_) in
-            self?.processCloudFiles()
-        }
-    }
+    private func zipWalletDatabase() throws -> URL {
+        let archiveName = fileName + ".zip"
 
-    private func processCloudFiles() {
-        if query.resultCount == 0 { return }
-        var fileItem: NSMetadataItem?
-        var fileURL: URL?
+        let tmpDirectory = try tempDirectory()
+        let archiveURL = tmpDirectory.appendingPathComponent(archiveName)
 
-        for item in query.results {
-            guard let item = item as? NSMetadataItem else { continue }
-            guard let fileItemURL = item.value(forAttribute: NSMetadataItemURLKey) as? URL else { continue }
-            if fileItemURL.lastPathComponent.contains(fileName) {
-                fileItem = item
-                fileURL = fileItemURL
-            }
+        if FileManager.default.fileExists(atPath: archiveURL.path) {
+            try FileManager.default.removeItem(atPath: archiveURL.path)
         }
 
-        let fileValues = try? fileURL!.resourceValues(forKeys: [URLResourceKey.ubiquitousItemIsUploadingKey])
-        if let fileUploaded = fileItem?.value(forAttribute: NSMetadataUbiquitousItemIsUploadedKey) as? Bool, fileUploaded == true, fileValues?.ubiquitousItemIsUploading == false {
-            notifyObservers(percent: 100, completed: true, error: nil)
-            progressValue = 0.0
-            inProgress = false
-            try? cleanTempDirectory()
-        } else if let error = fileValues?.ubiquitousItemUploadingError {
-            notifyObservers(percent: 0, completed: false, error: error)
-            progressValue = 0.0
-            inProgress = false
-        } else {
-            if let fileProgress = fileItem?.value(forAttribute: NSMetadataUbiquitousItemPercentUploadedKey) as? Double {
-                notifyObservers(percent: fileProgress, completed: false, error: nil)
-                progressValue = fileProgress
-            }
-        }
+        let sqlite3File = TariLib.databaseName.appending(".sqlite3")
+        try FileManager().zipItem(at: directory.appendingPathComponent(sqlite3File), to: archiveURL, compressionMethod: .deflate)
+        return archiveURL
     }
 
-    private func notifyObservers(percent: Double, completed: Bool, error: Error?) {
-        observers.allObjects.forEach {
-            if let object = $0 as? ICloudBackupObserver {
-                object.onUploadProgress(percent: percent, completed: completed, error: error)
-            }
-        }
-    }
+    private func encryptedZipWalletDatabase(keyString: String) throws -> URL {
```

Comment:
`password` may be a better parameter name.

Reply:
Try to make function names verbs. `zipAndEncryptWalletDatabase` or `getZippedAndEncryptedWalletDatabase`.